lisa/tools/lscpu.py

```diff
@@ -71,6 +91,29 @@ def get_cpu_type(self, force_run: bool = False) -> CpuType:
             return CpuType.Intel
         return CpuType.Intel
 
+    def get_cpu_info(self) -> List[CPUInfo]:
+        # `lscpu --extended=cpu,node,socket,cache` command return the
+        # cpu info in the format :
+        # CPU NODE SOCKET L1d:L1i:L2:L3
+        # 0    0        0 0:0:0:0
+        # 1    0        0 0:0:0:0
+        result = self.run("--extended=cpu,node,socket,cache").stdout
+        mappings_with_header = result.splitlines(keepends=False)
+        mappings = mappings_with_header[1:]
+        assert len(mappings) > 0
+        output: List[CPUInfo] = []
+        for item in mappings:
+            # Ignore first line as it contains column header
+            mapping = item.split()
+            cpu = int(mapping[0])
+            node = int(mapping[1])
+            socket = int(mapping[2])
+            cache = [int(c) for c in mapping[3].split(":")]
```

Comment:
It's better to use regular expression, instead of a couple of string operations. It's more straightforward. In our code base, there are many examples to reference.